suzieq/engines/pandas/interfaces.py

```diff
@@ -425,3 +470,145 @@ def _check_ipaddr(x, fld1, fld2, reason):
                             'peerHostname', 'peerIfname', 'check',
                             'assertReason', 'timestamp']] \
             .rename({'check': 'assert'}, axis=1)
+
+    def _add_portmode(self, df: pd.DataFrame):
+        """Add the switchport-mode i.e. acceess/trunk/routed'''
+
+        :param df[pd.Dataframe]: The dataframe to add vlanList to
+        :returns: original dataframe with portmode col added and filterd
+        """
+
+        if df.empty:
+            return df
+
+        conf_df = self._get_table_sqobj('devconfig') \
+            .get(namespace=df.namespace.unique().tolist(),
+                 hostname=df.hostname.unique().tolist())
+
+        devdf = self._get_table_sqobj('device') \
+            .get(namespace=df.namespace.unique().tolist(),
+                 hostname=df.hostname.unique().tolist(),
+                 columns=['namespace', 'hostname', 'os', 'status', 'vendor'])
+
+        if conf_df.empty:
+            df['portmode'] = np.where(df.ipAddressList.str.len() == 0,
+                                      'unknown',
+                                      'routed')
+            df['portmode'] = np.where(df.ip6AddressList.str.len() != 0,
+                                      'routed', df.portmode)
+
+            return df
+
+        pm_df = pd.DataFrame({'namespace': [], 'hostname': [],
+                              'ifname': [], 'portmode': []})
+
+        pm_list = []
+        for row in conf_df.itertuples():
+            # Check what type of device this is
+            # TBD: SONIC support
+            if not devdf.empty:
+                nos = devdf[(devdf.namespace == row.namespace) &
+                            (devdf.hostname == row.hostname)]['os'].tolist()[0]
+                if any(x in nos for x in ['junos', 'panos']):
+                    syntax = 'junos'
+                else:
+                    # The way we pull out Cumulus Linux conf is also ios-like
+                    syntax = 'ios'
+            else:
+                # Heuristics now
+                if '\ninterfaces {\n' in row.config or \
+                   'paloaltonetworks' in row.config:
+                    syntax = 'junos'
+                else:
+                    syntax = 'ios'
+            try:
+                conf = CiscoConfParse(row.config.split('\n'), syntax=syntax)
+            except Exception:  # pylint: disable=broad-except
+                break
```

Comment:
This for is highly dependent on the order due to the `break` at line 527.
Suppose you have 3 elements in conf_df: the first one is invalid (raise the CiscoConfParse exception) and the other 2 are valid. The result is that none of the is processes.
If the same 3 elements are in a different order [valid, invalid, valid], now only the first one is processed.
Using `continue` instead of break should fix this

Reply:
Another excellent catch, yes